syft/execution/protocol.py

```diff
@@ -141,8 +141,7 @@ def build(self):
         self.toggle_tracing(True)
         self.is_building = True
 
-        sorted_roles = [self.roles[name] for name in sorted(self.roles.keys())]
-        results = self.forward(*sorted_roles)
+        results = self.forward(*self.roles.values())
```

Comment:
Won't this assign values to args in the order of the dict (random)?

Reply:
In Python 3.7, the order is guaranteed to match insertion order: https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects

It's also true in Python 3.6, but it was provisional at that point: https://docs.python.org/3/whatsnew/3.6.html#new-dict-implementation

Reply:
Oh, really?
This goes against all I know about dicts 😆

Reply:
Oh, maybe they store metadata around the real data structure to keep this order when creating an iterator?

Reply:
Looks like they adopted a data structure that maintains the order of entries and uses a sparse index: https://mail.python.org/pipermail/python-dev/2012-December/123028.html